src/metrics.rs

```diff
@@ -1,5 +1,83 @@
 use std::alloc::{GlobalAlloc, Layout, System};
-use std::sync::atomic::{AtomicI64, Ordering};
+use std::sync::atomic::{AtomicBool, AtomicI64, Ordering};
+use std::time::Instant;
+
+use metrics::{describe_histogram, histogram, Unit};
+
+macro_rules! metric_const {
+    ($name: ident) => {
+        #[allow(non_upper_case_globals)]
+        pub(crate) const $name: &'static str = stringify!($name);
+    };
+}
+
+metric_const!(get_attempt_duration);
```

Comment:
how can we read these metrics?

Reply:
We can read them currently as they are not getting plumbed all the way to Julia, I am afraid the plumbing will have to come in a later PR.